cmd/gb-vendor/update.go

```diff
@@ -100,6 +90,11 @@ Flags:
 				return err
 			}
 
+			err = m.RemoveDependency(d)
```

Comment:
I am very sorry, I told you the wrong thing, m.RemoveDependency(d) checks that the dependency is present in the manifest. It should be above the call to DeduceRepo.